doc: add page about creating modules

authmillenon opened the pull request:
Moar doc :-)

PeterKietzmann commented on doc/doxygen/src/creating-modules.md:
```diff
@@ -0,0 +1,42 @@
+Pseudomodules are modules that do not have any code. Their main use cases are
+to provide base information for dependencies to other modules or information to
+the code base via the `MODULE_<MODULENAME>` macro. Pseudomodules can provide
+header files too, if need be. To create a pseudomodule just add its name to the
```
> if need to be

Stumbled upon that. How about "if there is a need for it" or similar

authmillenon replied:
"if need be" is a correct phrasing: http://www.oxforddictionaries.com/de/definition/englisch/if-need-be

Lotterleben replied:
+1, also “if need be” seems more concise to me.

PeterKietzmann commented on doc/doxygen/src/creating-modules.md:
```diff
@@ -0,0 +1,42 @@
+super-module.
+
+E.g. the @ref sys_shell module is implemented in `sys/shell` and defines its
+API in `sys/include/shell.h`.
```
Maybe it's nice to mention  which kinds of modules exist. Not like a complete list, but to inform that e.g. drivers are also modules

authmillenon replied:
Which one would that be? Also this page is primarily about *creating* modules, not using them ;-).

Lotterleben replied:
Having a (partial) list might help people to do their own digging for (good) examples...

authmillenon replied:
I'll add one if I get one. Up until now I got only: add some list ;-)

authmillenon replied:
After thinking about it, I think a better place for such list is the "Creating an application" page. I'm assuming here that most people that create a module at least worked a little bit with applications so they would have already seen that list. What do you guys think?

Lotterleben replied:
I'm starting to think that The List(tm) might be beneficial in both places :D

PeterKietzmann replied:
The only information that I was/am missing here was a hint that a module must not only be a system or kernel module but also a e.g. driver. IMO no need for a complete list here. The diver porting guide will also state that a driver might become a module

authmillenon replied:
Got it. Is it now better? @Lotterleben are you alright with this compromise?

PeterKietzmann commented:
I think it could be interesting to inform that the Makefile.dep handles some modules dependencies.

PeterKietzmann commented:
Again: Thanks Martine :-) !

Lotterleben commented:
Would it make sense to add a short sentence about what Modules actually are, maybe something along the lines of “Modules add a specific set of features to the RIOT base kernel” (or something o.o) or is that out of place here?

authmillenon commented:
They do not add to the kernel, but I can try to formulate a description:

> Modules in RIOT are well-defined units of code that provide a set of features to your application.

Lotterleben commented:
True. I like that sentence. :)

authmillenon commented:
Rebased and addressed comments (added section about definition of modules including clarifications that drivers and to an extend CPU and board ports [**disclaimer:** can someone like @haukepetersen fact-prove that ;-)] are modules, too; added section about `Makefile.dep`)

Lotterleben commented on doc/doxygen/src/creating-modules.md:
```diff
@@ -0,0 +1,63 @@
+Creating modules                                            {#creating-modules}
+================
+Modules in RIOT are well-defined units of code that provide a set of features
+to your application. This includes also drivers and to a further extend ports
```
I think it should be "to a certain extent"

authmillenon replied:
Fixed

authmillenon commented:
Fixed more comments. Now let's merge this so I can open a last minute release PR ;-)

Lotterleben commented:
ACK :) Could you squash please?

authmillenon commented:
Squashed.
